mlflow/langchain/__init__.py

```diff
@@ -314,13 +328,79 @@ def log_model(
 
 
 def _save_model(model, path):
-    model.save(path)
+    import langchain
+
+    model_data_path = os.path.join(path, _MODEL_DATA_FILE_NAME)
+    model_data_kwargs = {_MODEL_DATA_KEY: _MODEL_DATA_FILE_NAME}
+
+    if type(model) == langchain.chains.llm.LLMChain:
+        model.save(model_data_path)
+    elif type(model) == langchain.agents.agent.AgentExecutor:
+        if model.agent and model.agent.llm_chain:
```

Comment:
From https://python.langchain.com/en/latest/_modules/langchain/agents/initialize.html, it looks like users can pass any instance of `BaseLanguageModel` to LangChain's `initialize_agent` API. Does this mean that the LLM might not always be a chain? If it's not a chain, I think the current loading code will fail. Can we look into this and see if it's easy to support general instances of `BaseLanguageModel`? If it takes more than an hour or two to do that, we can just throw if the agent doesn't define a chain.

Reply:
Yes we check for that right now: https://github.com/mlflow/mlflow/pull/8297/files#diff-44b09327d1292c38e7ce251f66635a35c22720697ae1f39db6d4873d98109548R315

Maybe we can fast follow with adding other types or supporting `BaseLanguageModel` directly?

Reply:
Ah, missed the check. Thanks @sunishsheth2009 !